### DISCUSSION TITLE
Unmount CSS when component is unmounted

### DISCUSSION THREAD

**Discussion by roydukkey** [vuejs/rfcs#670] [General]
There are scenarios where leaving CSS attached is not optimal for performance. single-spa documents [Unmounting CSS](https://single-spa.js.org/docs/ecosystem-css/#unmounting-css) ([canonical](https://github.com/single-spa/single-spa.js.org/blob/89c97f1933e306e73a6ff50c533ef2c8cdb341e2/versioned_docs/version-6.x/ecosystem-css.md?plain=1#L205-L235)). It would be nice to have options within Vue to handle removing `<style />` and `<link />` elements from the DOM when the component unmounts.

For SFCs, I think the feature could be fairly straight forward to define, but I'd imagine that the feature might not exactly be implemented in Vue core itself, perhaps the SFC compiler and/or bundlers. Just not sure myself.

```vue
<script setup>
import './style.scss?forget';
</script>

<style lang="scss" scoped forget></style>
```

Not sure the best term, but `forget` seems simple and descriptive.

**Comment by ferferga** [comment]
I don't understand your problem. The CSS in SFCs are loaded globally, with specific selectors that matches the components scope id. They're no per component.
